odoo/addons/base/models/ir_attachment.py

```diff
@@ -181,19 +181,25 @@ def _gc_file_store(self):
             cr.rollback()
             return False
 
-        self._gc_file_store_unsafe()
+        done, remaining = self._gc_file_store_unsafe()
 
         # commit to release the lock
         cr.commit()
 
-    def _gc_file_store_unsafe(self):
+        return done, remaining
+
+    def _gc_file_store_unsafe(self, limit=10_000):
```

Comment:
Nitpick: You could set the limit to `cr.IN_MAX` and remove the loop that splits the checklist.